### PR TITLE
Fix memleak in CapturePix

### PR DESCRIPTION
**Pull request by vlolteanu**
If there's some error when writing the screenshot to the file, the buffer does not get freed.

(As a fun fact, this is literally the first hit that my IDE yielded when searching for DiabloAllocPtr.)

I was conservative with this commit, and used the DiabloAllocPtr and mem_free_dbg macros. Going forward, should we still use them in cases where we can make memory management automatic? (The network code, for example, does not use them.)

### CODE REVIEW

**Comment by glebm** [comment]
Do we actually want a `DiabloUniqPtr` type? Perhaps we should just use `std::unique_ptr`, and use proper tools to track allocations when needed.

**Comment by AJenbo** [comment]
Agreed, throw it away

**Comment by vlolteanu** [comment]
Done.

**Comment by vlolteanu** [comment]
DiabloMakeUnique might still be useful for e.g. TMsg, where memory is allocated beyond a struct's boundary, expecting some more data to follow.

File: Source/engine.h [hunk 1]
```diff
@@ -134,6 +135,18 @@ struct ActorPosition {
 #define mem_free_dbg(PTR) \
 	std::free(PTR)
 
+template <typename T>
+std::unique_ptr<T> DiabloMakeUnique(size_t size)
```
Comment (glebm) [inline Source/engine.h:139]: Unused, please remove

File: Source/capture.cpp [hunk 2]
```diff
@@ -116,16 +116,15 @@ static BYTE *CaptureEnc(BYTE *src, BYTE *dst, int width)
 static bool CapturePix(const CelOutputBuffer &buf, std::ofstream *out)
 {
 	int width = buf.w();
-	BYTE *pBuffer = (BYTE *)DiabloAllocPtr(2 * width);
+	std::unique_ptr<BYTE []> pBuffer = std::make_unique<BYTE[]>(2 * width);
```
Comment (glebm) [inline Source/capture.cpp:119]:
```suggestion
	auto pBuffer = std::make_unique<BYTE[]>(2 * width);
```

**Review by glebm — APPROVED** [review]

**Review by AJenbo — APPROVED** [review]

**Comment by AJenbo** [comment]
> I was conservative with this commit

12 commits... I'll just squash this one any way.

**Pull request merged**